### ISSUE TITLE
Non-conventional use of ng-init

### ISSUE THREAD

**Issue by androidfred** [meanjs/mean#325]
From Google group discussion: https://groups.google.com/forum/#!topic/meanjs/cO5AAAOgVAY

Angular docs for ngInit (https://docs.angularjs.org/api/ng/directive/ngInit#!) say:

"The only appropriate use of ngInit is for aliasing special properties of ngRepeat, as seen in the demo below. Besides this case, you should use controllers rather than ngInit to initialize values on a scope.

But e.g. in list-articles.client.view.html it is being used as below:

    <section data-ng-controller="ArticlesController" data-ng-init="find()">

This appears to be a part of the boilerplate that breaks convention and unless there is good reason for that it should arguably be fixed. I'd be happy to do a pull request and fix it myself, though I'm not sure if data-ng-init="find()" should simply be removed or replaced by something else.

**Comment by androidfred** [comment]
In case anyone cares, take a look at the controllers, views and karma tests in this module to see how I changed things: https://github.com/androidfred/ideit/tree/master/public/modules/ideas

TLDR I split the controller, run the initialization methods after declaring them in the controllers, got rid of ng-init and changed the tests slightly.

I'd be happy to do a PR but feedback and agreement from you guys is required first.

**Comment by ikhos** [comment]
I find the proposed changes better than using ng-init, but kind of cumbersome. Couldn't this be delegated to a function that checks the current $state? Will this affect performance in any way?

Like so:

```
$scope.$on('$stateChangeSuccess', function(event, toState){
  switch (toState) {
    case 'view':
      $scope.findOne();
      break;
    case 'list':
      $scope.findOne();
      break;
    default:
      //Default logic
  }
});
```

**Comment by ilanbiala** [comment]
@ikhos I'm not really a fan of doing stuff in the event, because that makes more work for people who want to switch routing and if we decide to move to angular/router in the future.

**Comment by androidfred** [comment]
Thanks for your attention.

As Angular docs say you should use controllers rather than ngInit to initialize values on a scope, and that a controller should contain only the business logic needed for a single view, I hope both these issues are addressed by some other means.

**Comment by mleanos** [comment]
@androidfred Have you explored implementing a resolve on your ui routes? This could help keep your controllers clean of these types of data initializing. I've seen this approach advocated. Personally, I prefer having an init() function in my controllers. It's easier for me to keep track of what data is being loaded when the controller's initialize; without the need to track it down in my route configurations. 

I haven't implemented many resolves on my ui routes, as I'm not sure of the performance costs since the route won't load until the resolve finishes.

**Comment by rhutchison** [comment]
Resolved by #1126 

Please close if this is no longer an issue.

**Comment by androidfred** [comment]
Closing as requested.
